Unable to configure JDBC idempotent consumers

If I try to do:

```xml
    <jee:jndi-lookup id="dataSource" jndi-name="java:jboss/datasources/ExampleDS" />

    <spring:bean id="messageIdRepository" class="org.apache.camel.processor.idempotent.jdbc.JdbcMessageIdRepository">
        <spring:constructor-arg ref="dataSource" />
    </spring:bean>
```

I get:

```
Caused by: java.lang.NoClassDefFoundError: org/springframework/transaction/PlatformTransactionManager
        at java.lang.Class.getDeclaredConstructors0(Native Method)
        at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
        at java.lang.Class.getDeclaredConstructors(Class.java:2020)
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:149)
        ... 20 more
Caused by: java.lang.ClassNotFoundException: org.springframework.transaction.PlatformTransactionManager from [Module "org.apache.camel.component.sql:main" from local module loader @704d6e83 (finder: local module finder @43a0cee9 
```

--- a/feature-pack/src/main/resources/modules/system/layers/fuse/org/apache/camel/component/sql/main/module.xml
+++ b/feature-pack/src/main/resources/modules/system/layers/fuse/org/apache/camel/component/sql/main/module.xml
@@ -6,6 +6,7 @@
   <dependencies>
     <module name="javax.api" />
     <module name="org.slf4j" />
+    <module name="org.springframework.tx" />
     <module name="javax.xml.bind.api" />
     <module name="org.apache.camel.core" />
     <module name="org.springframework.jdbc" />

--- a/itests/standalone/basic/src/main/java/org/wildfly/camel/test/sql/SQLIntegrationTest.java
+++ b/itests/standalone/basic/src/main/java/org/wildfly/camel/test/sql/SQLIntegrationTest.java
@@ -28,8 +28,11 @@ import javax.sql.DataSource;
 
 import org.apache.camel.CamelContext;
 import org.apache.camel.PollingConsumer;
+import org.apache.camel.ProducerTemplate;
 import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.mock.MockEndpoint;
 import org.apache.camel.impl.DefaultCamelContext;
+import org.apache.camel.processor.idempotent.jdbc.JdbcMessageIdRepository;
 import org.jboss.arquillian.container.test.api.Deployer;
 import org.jboss.arquillian.container.test.api.Deployment;
 import org.jboss.arquillian.junit.Arquillian;
@@ -114,6 +117,39 @@ public class SQLIntegrationTest {
             Assert.assertEquals("SA", result);
         } finally {
             deployer.undeploy(CAMEL_SQL_CDI_ROUTES_JAR);
+        }
+    }
+
+    @Test
+    public void testSqlIdempotentConsumer() throws Exception {
+        Assert.assertNotNull("DataSource not null", dataSource);
+
+        final JdbcMessageIdRepository jdbcMessageIdRepository = new JdbcMessageIdRepository(dataSource, "myProcessorName");
+
+        CamelContext camelctx = new DefaultCamelContext();
+        camelctx.addRoutes(new RouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                from("direct:start")
+                .idempotentConsumer(simple("${header.messageId}"), jdbcMessageIdRepository)
+                .to("mock:result");
+            }
+        });
+
+        camelctx.start();
+        try {
+            MockEndpoint mockEndpoint = camelctx.getEndpoint("mock:result", MockEndpoint.class);
+            mockEndpoint.expectedMessageCount(1);
+
+            // Send 5 messages with the same messageId header. Only 1 should be forwarded to the mock:result endpoint
+            ProducerTemplate template = camelctx.createProducerTemplate();
+            for (int i = 0; i < 5; i++) {
+                template.requestBodyAndHeader("direct:start", null, "messageId", "12345");
+            }
+
+            mockEndpoint.assertIsSatisfied();
+        } finally {
+            camelctx.stop();
         }
     }
 }

--- a/modules/etc/generated/wildfly/modules/system/layers/fuse/org/apache/camel/component/sql/main/module.xml
+++ b/modules/etc/generated/wildfly/modules/system/layers/fuse/org/apache/camel/component/sql/main/module.xml
@@ -6,6 +6,7 @@
   <dependencies>
     <module name="javax.api" />
     <module name="org.slf4j" />
+    <module name="org.springframework.tx" />
     <module name="javax.xml.bind.api" />
     <module name="org.apache.camel.core" />
     <module name="org.springframework.jdbc" />

--- a/modules/etc/smartics/camel-modules.xml
+++ b/modules/etc/smartics/camel-modules.xml
@@ -717,6 +717,7 @@
         <dependencies>
             <module name="javax.api"/>
             <module name="org.slf4j"/>
+            <module name="org.springframework.tx" />
         </dependencies>
     </module>
     

